src/ray/common/client_connection.h

```diff
@@ -193,11 +214,105 @@ class ClientConnection : public ServerConnection<T> {
   std::vector<uint8_t> read_message_;
 };
 
+/// \typename SimpleConnection
+///
+/// A generic type representing a client connection on a server. In addition to
+/// writing messages to the client, like in ServerConnection, this typename can
+/// also be used to process messages asynchronously from client.
+template <typename T>
+class SimpleConnection {
```

Comment:
How about rename it to `RayletConnection` or `ThreadSafeConnection`, `SimpleConnection` seems not very intuitive and doesn't really represent what this class does.

My understand is this thread-safe behavior is only used for raylet client, it's not used by  object manager nor node manager, and thus it is probably more appropriate to be placed to raylet client code instead of in common folder. Suggest to move it from here to raylet_client.h or a new raylet_connection.h.

Reply:
Yes, I should rename it. This client could be used for general clients where thread safe matters, so I still want to keep it in the common directory.